### PR TITLE
Add a mission about how to display a keystone in Hai space

### PR DESCRIPTION
**Pull request by roadrunner56**
**Content (Artwork / Missions / Jobs)**

## Summary
In #9753, @MasterOfGrey mentioned wanting to have a short introductory mission that registered the player as someone who carried a keystone, and told the player how to display their keystone to potential customers. This would allow them to take jobs and missions that required a keystone. This is a short mission that does exactly that.

## Testing Done
lol

## Save File
N/A

### CODE REVIEW

File: data/hai/hai missions.txt [hunk 1]
```diff
@@ -1932,3 +1932,29 @@ mission "Station Exploration Abandon Ship"
...
+			has "outfit: Tree Skeleton Key Stone"
+		has "First Contact: Hai: offered"
+	on offer
+		conversation
+			`Upon landing, you receive a message from the local port authority. "Hello, Captain <last>. We have been informed that you have recently installed a keystone, or its equivalent, onto your ship. We have registered you with the job board, which will allow you to take passengers that specifically request a keystone-equipped ship." Once the message ends, you can hear a quiet tapping on the door of your ship's hatch. You turn to see a short Hai in a spaceport uniform peeking through your window. He waves and smiles.`
+			choice
+				`	(Open the door.)`
+					goto open
+				`	(Ignore him.)`
+			`	Your deliberate disregard for the Hai at your door doesn't appear to stop him. "I saw you had a keystone installed!" His voice is muffled despite his shouting, but you can still hear what he says. "To advertise your ship's keystone in the spaceport, my advice is to mount it on the inside of your ship's main door. When you have it open, passerby can see you have a keystone and ask for transportation. You meet some funny people that way. Good luck, Captain!"`
```
Comment (bene-dictator) [inline data/hai/hai missions.txt:1956]:
```suggestion
			`	Your deliberate disregard for the Hai at your door doesn't appear to stop him. "I saw you had a keystone installed!" His voice is muffled despite his shouting, but you can still hear what he says. "To advertise your ship's keystone in the spaceport, my advice is to mount it on the inside of your ship's main door. When you have it open, passersby can see you have a keystone and ask for transportation. You meet some funny people that way. Good luck, Captain!"`
```

File: data/hai/hai missions.txt [hunk 2]
```diff
@@ -1932,3 +1932,29 @@ mission "Station Exploration Abandon Ship"
...
+				decline
+			label open
+			`	The Hai wastes no time in getting to the point; the second you open your door, he begins to speak. "I saw you had a keystone installed! To advertise your ship's keystone in the spaceport, my advice is to mount it on the inside of your ship's main door. When you have it open, passerby can see you have a keystone and ask for transportation. You meet some funny people that way. Good luck, Captain!"`
```
Comment (bene-dictator) [inline data/hai/hai missions.txt:1959]:
```suggestion
			`	The Hai wastes no time in getting to the point; the second you open your door, he begins to speak. "I saw you had a keystone installed! To advertise your ship's keystone in the spaceport, my advice is to mount it on the inside of your ship's main door. When you have it open, passersby can see you have a keystone and ask for transportation. You meet some funny people that way. Good luck, Captain!"`
```

**Review by bene-dictator — CHANGES_REQUESTED** [review]

File: data/hai/hai missions.txt [hunk 3]
```diff
@@ -1932,3 +1932,29 @@ mission "Station Exploration Abandon Ship"
...
+	to offer
+		or
+			has "outfit: Quantum Keystone"
+			has "outfit: Quantum Key Stone"
+			has "outfit: Shield Beetle Pendant"
+			has "outfit: Tree Skeleton Key Stone"
+		has "First Contact: Hai: offered"
+	on offer
+		conversation
+			`Upon landing, you receive a message from the local port authority. "Hello, Captain <last>. We have been informed that you have recently installed a keystone, or its equivalent, onto your ship. We have registered you with the job board, which will allow you to take passengers that specifically request a keystone-equipped ship." Once the message ends, you can hear a quiet tapping on the door of your ship's hatch. You turn to see a short Hai in a spaceport uniform peeking through your window. He waves and smiles.`
```
Comment (Hurleveur) [inline data/hai/hai missions.txt:1951]: the jobs are intended to only accept the actual hai keystone, iirc?

Comment (roadrunner56) [reply data/hai/hai missions.txt]: Changed to have the message mention this. Perhaps we should add jobs that pay less but are less picky about what kind of keystone the player has?

**Comment by roadrunner56** [comment]
Another thought regarding jobs that require keystones: perhaps we should have jobs that only require keystones, and don't need luxury accommodations? This would account for Hai that are superstitious but not necessarily wealthy.

**Comment by bene-dictator** [comment]
That would be a good idea, though I thought that was the case already.

**Comment by Anarchist2** [comment]
This feels largely redundant with the outfit description we already have. Dedicating an entire conversation to that detail makes it stick out awkwardly.

**Comment by Hecter94** [comment]
I'd prefer this over a mention in the description if having both would be redundant.

**Comment by roadrunner56** [comment]
> This feels largely redundant with the outfit description we already have. Dedicating an entire conversation to that detail makes it stick out awkwardly.

I mentioned this to @MasterOfGrey in #9753 (https://github.com/endless-sky/endless-sky/pull/9753#issuecomment-1950974573) and he thought the idea was appropriate. I'd like to hear Grey's thoughts on this one.

**Comment by Amazinite** [comment]
I'm in agreement with @Anarchist2 on this. This mission is effectively throwing a tutorial in the player's face for how to use keystones and putting up a big sign that says, "Hey, this is important! Make sure you keep this on!" While we're not entirely outside of including such tutorial-like missions in the game (we do have missions to warn the player about how a high combat rating will spawn bounty hunters, a high pirate attractiveness will spawn raids, and if you ignore the spaceport for a long time we'll call you out for that too), I think that the less such missions we have, the better. Also of note is that those examples that I provided are early-game stuff; they all offer in human space to remind the player of important gameplay mechanics (two of which, bounty hunters and pirate raids, are only even in effect in human space). I don't think we should still be trying to tutorialize stuff once the player has jumped out of human space. Leaving human space should be the player's opportunity to explore on their own and find their own way in the galaxy.

File: data/hai/hai missions.txt [hunk 4]
```diff
@@ -1932,3 +1932,29 @@ mission "Station Exploration Abandon Ship"
...
+	on offer
+		conversation
+			`Upon landing, you receive a message from the local port authority. "Hello, Captain <last>. We have been informed that you have recently installed a keystone, or its equivalent, onto your ship. We have registered you with the job board, which will allow you to take passengers that specifically request a keystone-equipped ship. However, be aware that most passengers will not board ships equipped with non-standard keystones." Once the message ends, you can hear a quiet tapping on the door of your ship's hatch. You turn to see a short Hai in a spaceport uniform peeking through your window. He waves and smiles.`
+			choice
+				`	(Open the door.)`
+					goto open
+				`	(Ignore him.)`
+			`	Your deliberate disregard for the Hai at your door doesn't appear to stop him. "I saw you had a keystone installed!" His voice is muffled despite his shouting, but you can still hear what he says. "To advertise your ship's keystone in the spaceport, my advice is to mount it on the inside of your ship's main door. When you have it open, passersby can see you have a keystone and ask for transportation. You meet some funny people that way. Good luck, Captain!"`
```
Comment (Amazinite) [inline data/hai/hai missions.txt:1956]: This is funny, though.

**Comment by roadrunner56** [comment]
Seeing as this PR's purpose is made redundant by the keystone description as well as #9753, I'm closing this for now.

**Comment by MasterOfGrey** [comment]
Yeah, discounting the last line, I would have described this as a good start for what I was thinking of.
But this isn't all there for that.

The missions I referred to in my example were reputation and trigger locked, with more than one step and an actual need to go somewhere, and when you finish them you're effectively whitelisted for access to new jobs.

**Pull request closed without merging**